File: framework/languages/node/samples/TicTacToe.Ts/Server/Play/Infrastructure/ZLink/Spots/TicTacToeGameSpot/Handlers/tictactoe-game-timer-handler.ts
import type {
  ZLinkSpotTimerHandler,
  ZLinkTimerTick
} from '@zlink-systems/framework';
import { zlinkSpotTimerHandler } from '@zlink-systems/nestjs';
import { ZLinkTimerOverrunPolicy } from '@zlink-systems/framework';
import { TicTacToeGameSpot } from '../tictactoe-game-spot';

@zlinkSpotTimerHandler({
  spot: () => TicTacToeGameSpot,
  name: 'game-tick',
  periodMs: 1000,
  options: {
    overrunPolicy: ZLinkTimerOverrunPolicy.DelayNextTick,
    stopOnUnhandledException: true
  }
})
// --8<-- [start:doc-timer-handler]
class TicTacToeGameTimerHandler implements ZLinkSpotTimerHandler<TicTacToeGameSpot> {
  async handle(spot: TicTacToeGameSpot, tick: ZLinkTimerTick): Promise<void> {
    void tick;
    await spot.tick();
  }
}
// --8<-- [end:doc-timer-handler]

export { TicTacToeGameTimerHandler };

File: framework/languages/java/samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/play/infrastructure/zlink/spots/tictactoegamespot/handlers/TicTacToeGameTimerHandler.java
package systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.handlers;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;
import systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.TicTacToeGame;

// --8<-- [start:doc-timer-handler]
public final class TicTacToeGameTimerHandler implements ZLinkSpotTimerHandler<TicTacToeGame> {
    @Override
    public java.util.concurrent.CompletionStage<Void> handle(
        TicTacToeGame spot,
        ZLinkTimerTick tick) {
        return spot.tick();
    }
}
// --8<-- [end:doc-timer-handler]
